app/assets/javascripts/components/AccountMenu/index.tsx

```diff
@@ -0,0 +1,163 @@
+      async () => {
+        setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+      },
+      ApplicationEvent.ProtectionSessionExpiryDateChanged
+    );
+
+    return () => {
+      removeKeyStatusChangedObserver();
+      removeProtectionSessionExpiryDateChangedObserver();
+    };
+  }, [application, getProtectionsDisabledUntil]);
+
+  return (
+    <div className="sn-component">
+      <div id="account-panel" className="sk-panel">
+        <div className="sk-panel-header">
+          <div className="sk-panel-header-title">Account</div>
+          <a className="sk-a info close-button" onClick={closeAccountMenu}>Close</a>
+        </div>
+        <div className="sk-panel-content">
+          <Authentication
+            application={application}
+            server={server}
+            setServer={setServer}
+            closeAccountMenu={closeAccountMenu}
+            notesAndTagsCount={notesAndTagsCount}
+            showLogin={showLogin}
+            setShowLogin={setShowLogin}
+            showRegister={showRegister}
+            setShowRegister={setShowRegister}
+            user={user}
+          />
+          {!showLogin && !showRegister && (
+            <div>
+              {user && (
+                <User
+                  email={user.email}
+                  server={server}
+                  appState={appState}
+                  application={application}
+                  closeAccountMenu={closeAccountMenu}
+                />
+              )}
+              <Encryption
+                isEncryptionEnabled={isEncryptionEnabled}
+                notesAndTagsCount={notesAndTagsCount}
+                encryptionStatusString={encryptionStatusString}
+              />
+              {hasProtections && (
+                <Protections
+                  application={application}
+                  protectionsDisabledUntil={protectionsDisabledUntil}
+                />
+              )}
+              <PasscodeLock
+                application={application}
+                setEncryptionStatusString={setEncryptionStatusString}
+                setIsEncryptionEnabled={setIsEncryptionEnabled}
+                setIsBackupEncrypted={setIsBackupEncrypted}
+              />
+              <DataBackup
+                application={application}
+                isBackupEncrypted={isBackupEncrypted}
+                isEncryptionEnabled={isEncryptionEnabled}
+                setIsBackupEncrypted={setIsBackupEncrypted}
+              />
+              <ErrorReporting appState={appState} />
+            </div>
+          )}
+        </div>
+        <ConfirmSignoutContainer application={application} appState={appState} />
+        <Footer
+          appState={appState}
+          application={application}
+          showLogin={showLogin}
+          setShowLogin={setShowLogin}
+          showRegister={showRegister}
+          setShowRegister={setShowRegister}
+          user={user}
```

Comment:
Same comment probably applies here, if there are shared variables between Authentication, Footer and the parent component we could move them to appState

Reply:
Sure, done. Regarding `user` - I decided to move it from `account_menu_state`, as we already can get it by `application.getUser()`. All the others are moved to `account_menu_state`.